diff --git a/include/petsc/private/cupminterface.hpp b/include/petsc/private/cupminterface.hpp
index a9b13fcfac7..e6ea14bc1d2 100644
--- a/include/petsc/private/cupminterface.hpp
+++ b/include/petsc/private/cupminterface.hpp
@@ -242,23 +242,14 @@ struct InterfaceImpl<DeviceType::CUDA> : InterfaceBase<DeviceType::CUDA> {
   // memory management
   PETSC_CUPM_ALIAS_FUNCTION(cupmFree, cudaFree)
   PETSC_CUPM_ALIAS_FUNCTION(cupmMalloc, cudaMalloc)
-  #if PETSC_PKG_CUDA_VERSION_GE(11, 2, 0)
-  PETSC_CUPM_ALIAS_FUNCTION(cupmFreeAsync, cudaFreeAsync)
-  PETSC_CUPM_ALIAS_FUNCTION(cupmMallocAsync, cudaMallocAsync)
-  #else
   PETSC_CUPM_ALIAS_FUNCTION_GOBBLE(cupmFreeAsync, cudaFree, 1)
   PETSC_CUPM_ALIAS_FUNCTION_GOBBLE(cupmMallocAsync, cudaMalloc, 1)
-  #endif
   PETSC_CUPM_ALIAS_FUNCTION(cupmMemcpy, cudaMemcpy)
   PETSC_CUPM_ALIAS_FUNCTION(cupmMemcpyAsync, cudaMemcpyAsync)
   PETSC_CUPM_ALIAS_FUNCTION(cupmMallocHost, cudaMallocHost)
   PETSC_CUPM_ALIAS_FUNCTION(cupmFreeHost, cudaFreeHost)
   PETSC_CUPM_ALIAS_FUNCTION(cupmMemset, cudaMemset)
-  #if PETSC_PKG_CUDA_VERSION_GE(11, 2, 0)
-  PETSC_CUPM_ALIAS_FUNCTION(cupmMemsetAsync, cudaMemsetAsync)
-  #else
   PETSC_CUPM_ALIAS_FUNCTION_GOBBLE(cupmMemsetAsync, cudaMemset, 1)
-  #endif
   PETSC_CUPM_ALIAS_FUNCTION(cupmMemcpy2D, cudaMemcpy2D)
   PETSC_CUPM_ALIAS_FUNCTION_GOBBLE(cupmMemcpy2DAsync, cudaMemcpy2DAsync, 1)
   PETSC_CUPM_ALIAS_FUNCTION(cupmMemset2D, cudaMemset2D)
diff --git a/src/sys/objects/device/impls/cupm/cupmcontext.hpp b/src/sys/objects/device/impls/cupm/cupmcontext.hpp
index e21262c6237..8187fd7b6c6 100644
--- a/src/sys/objects/device/impls/cupm/cupmcontext.hpp
+++ b/src/sys/objects/device/impls/cupm/cupmcontext.hpp
@@ -246,8 +246,8 @@ inline PetscErrorCode DeviceContext<T>::initialize(PetscDevice device) noexcept
     cupmMemPool_t mempool;
 
     initialized_ = true;
-    PetscCallCUPM(cupmDeviceGetMemPool(&mempool, static_cast<int>(device->deviceId)));
-    PetscCallCUPM(cupmMemPoolSetAttribute(mempool, cupmMemPoolAttrReleaseThreshold, &threshold));
+    // PetscCallCUPM(cupmDeviceGetMemPool(&mempool, static_cast<int>(device->deviceId)));
+    // PetscCallCUPM(cupmMemPoolSetAttribute(mempool, cupmMemPoolAttrReleaseThreshold, &threshold));
     blashandles_.fill(nullptr);
     solverhandles_.fill(nullptr);
     PetscCall(PetscRegisterFinalize(finalize_));